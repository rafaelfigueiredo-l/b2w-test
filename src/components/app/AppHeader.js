import React, { Component } from "react";
import styled from "styled-components";
import Logo from "../../assets/images/logo.png";


const AppHeader = (props) => {

  return (
    <>
      <Header><a href="#"><img src={Logo}/></a></Header>
    </>
  );
};

const Header = styled.header`
  display: flex;
  align-items: center;
  justify-content: center;
  padding:20px 0;
    a{
        text-align:center;
        width:100%;
    }
    img{
        max-width:200px;
        width:100%;
    }
`;

export default AppHeader;
