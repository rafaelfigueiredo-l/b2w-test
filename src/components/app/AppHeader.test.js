import AppHeader from './AppHeader';
import Logo from "../../assets/images/logo.png";

describe('<AppHeader />', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<AppHeader />, div);
        ReactDOM.unmountComponentAtNode(div);
      });

    it('Snapshot testing', () => {
        const wrapper = shallow(<AppHeader />);
        expect(wrapper).toMatchSnapshot();
    });

    it('Component should have logo', () => {
        const wrapper = shallow(<AppHeader />);
        expect(wrapper.find("img").prop("src")).toEqual(Logo);
    });
});