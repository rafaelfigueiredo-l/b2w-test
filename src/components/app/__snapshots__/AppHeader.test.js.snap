// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppHeader /> Snapshot testing 1`] = `
<Fragment>
  <styled.header>
    <a
      href="/#"
    >
      <img
        alt="Star Wars Logo"
        src="logo.png"
      />
    </a>
  </styled.header>
</Fragment>
`;
